pkgs/applications/window-managers/x-create-mouse-void/default.nix

```diff
@@ -0,0 +1,35 @@
+{ stdenv, lib, xorg, fetchFromGitHub }:
+stdenv.mkDerivation rec {
+  pname = "x-create-mouse-void";
+  version = "0.1";
+
+  src = fetchFromGitHub {
+    owner = "cas--";
+    repo = "XCreateMouseVoid";
+    rev = "0.1";
+    sha256 = "151pv4gmzz9g6nd1xw94hmawlb5z8rgs1jb3x1zpvn3znd7f355c";
+  };
+
+  buildInputs = [ xorg.libX11 ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -pv $out/bin
+    cp -av XCreateMouseVoid $out/bin/x-create-mouse-void
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/cas--/XCreateMouseVoid";
+    description =
+      "Creates an undecorated black window and prevents the mouse from entering that window";
+    platforms = platforms.unix;
+    license = ''
```

Comment:
The package does not really have a license so we must set license.unfree.

Reply:
Applied. 

There is a license granted, but I’m not sure what our definition of “free” is. Any license explicitly listed by FSF/GNU as libre?

Do you think “unfree-redistributable” would also work?

Reply:
free licenses must at least allow us to freely distribute the source code and compilation.



> Do you think “unfree-redistributable” would also work?

I think that would work, too.

Reply:
Thanks! Amended…

Reply:
https://github.com/cas--/XCreateMouseVoid/blob/master/XCreateMouseVoid.c#L1-L6

unfree redistributable will do for now. The comment suggests they're pretty flexible but you can always email patrick & cas-- and maybe get them to explicitly agree to something like GPLv3 or WTFPL :shrug:

Reply:
also welcome to nixpkgs @eigengrau 

I noticed you maintain this package on the AUR, is this a sign you'll maintain more stuff in nixpkgs going forward? :rocket: